tokio-udp/src/socket.rs

```diff
@@ -52,13 +53,19 @@ impl UdpSocket {
         self.io.get_ref().connect(*addr)
     }
 
-    #[deprecated(since = "0.1.2", note = "use poll_send instead")]
-    #[doc(hidden)]
-    pub fn send(&mut self, buf: &[u8]) -> io::Result<usize> {
-        match self.poll_send(buf)? {
-            Async::Ready(n) => Ok(n),
-            Async::NotReady => Err(io::ErrorKind::WouldBlock.into()),
-        }
+    /// Returns a future that sends data on the socket to the remote address to which it is connected.
+    /// On success, the future will resolve to the number of bytes written.
+    ///
+    /// The [`connect`] method will connect this socket to a remote address. The future
+    /// will resolve to an error if the socket is not connected.
+    ///
+    /// [`connect`]: #method.connect
+    ///
+    /// # Panics
```

Comment:
I think we can drop this, and this should be a detail on the future docs it self. Most users will be using this within an async context anyways. This was a much bigger issue with futures 0.1.